packages/react-tinacms-editor/src/tinacms-inline/inline-wysiwyg.tsx

```diff
@@ -33,10 +33,27 @@ export function InlineWysiwyg({
   name,
   children,
   focusRing = true,
+  imageProps: passedInImageProps,
   ...wysiwygProps
 }: InlineWysiwygFieldProps) {
   const cms = useCMS()
 
+  const imageProps: ImageProps = React.useMemo(() => {
+    return {
+      async upload(files: File[]) {
+        const allMedia = await cms.media.store.persist(
+          files.map(file => ({
+            directory: passedInImageProps?.directory || '',
+            file,
+          }))
+        )
+
+        return allMedia.map(media => `${media.directory}${media.filename}`)
+      },
+      ...passedInImageProps,
+    }
+  }, [cms.media.store])
```

Comment:
```suggestion
  }, [cms.media.store, passedInImageProps])
```
should we add this here?

Reply:
Maybe. It would probably make the hook rerun on every render

Reply:
maybe we're memoizing the wrong thing then

```ts
const defaultUpload = React.useCallback((files: File[]) => {
  //...
}, [cms.media.store])

const imageProps = {
  upload: defaultUpload,
  ...passedInImageProps
}
```